packages/sanity/src/core/studio/addOnDataset/setup/CommentsSetupProvider.tsx

```diff
@@ -51,60 +54,56 @@ export function CommentsSetupProvider(props: CommentsSetupProviderProps) {
     [originalClient, projectId],
   )
 
-  const handleRunSetup = useCallback(
-    async (comment: CommentPostPayload) => {
-      setIsRunningSetup(true)
-
-      // Before running the setup, we check if the addon dataset already exists.
-      // The addon dataset might already exist if another user has already run
-      // the setup, but the current user has not refreshed the page yet and
-      // therefore don't have a client for the addon dataset yet.
-      try {
-        const addonDatasetName = await getAddonDatasetName()
-
-        if (addonDatasetName) {
-          const client = handleCreateClient(addonDatasetName)
-          setAddonDatasetClient(client)
-          await client.create(comment)
-          setIsRunningSetup(false)
-          return
-        }
-      } catch (_) {
-        // If the dataset does not exist we will get an error, but we can ignore
-        // it since we will create the dataset in the next step.
-      }
+  const handleRunSetup = useCallback(async (): Promise<SanityClient | null> => {
+    setIsRunningSetup(true)
+
+    // Before running the setup, we check if the addon dataset already exists.
+    // The addon dataset might already exist if another user has already run
+    // the setup, but the current user has not refreshed the page yet and
+    // therefore don't have a client for the addon dataset yet.
+    try {
+      const addonDatasetName = await getAddonDatasetName()
 
-      try {
-        // 1. Create the addon dataset
-        const res = await originalClient.withConfig({apiVersion: API_VERSION}).request({
-          uri: `/comments/${dataset}/setup`,
-          method: 'POST',
-        })
-
-        const datasetName = res?.datasetName
-
-        // 2. We can't continue if the addon dataset name is not returned
-        if (!datasetName) {
-          setIsRunningSetup(false)
-          return
-        }
-
-        // 3. Create a client for the addon dataset and set it in the context value
-        //    so that the consumers can use it to execute comment operations and set up
-        //    the real time listener for the addon dataset.
-        const client = handleCreateClient(datasetName)
+      if (addonDatasetName) {
+        const client = handleCreateClient(addonDatasetName)
         setAddonDatasetClient(client)
+        setIsRunningSetup(false)
+        return client
+      }
+    } catch (_) {
```

Comment:
I'm afraid I'm not as close to this as you are, but is there a specific error we can catch here? I'm just concerned about valid errors being ignored and further work being done, getting into a strange state.

The additional error catching below might suffice, but was curious if we could get more specificity here.